src/Propel/Common/Config/Loader/FileLoader.php

```diff
@@ -61,6 +61,8 @@ public function __construct(FileLocatorInterface $locator = null)
      * Replaces parameter placeholders (%name%) by their values for all parameters.
      *
      * @param array $configuration The configuration array to resolve
+     *
+     * @return array|void
```

Comment:
You are wrong about this.
Void means that this method cannot by definition return anything, you "may" not check on the return result, as it makes no sense.
So if something returns anything not null, then it always must always return null then (instead of void) as alternative. Because then you actually do check on the return type and `!== null` etc.
As the block most cleary explains, there is no `something|void` from a logical stand point.
PHP7 typehinting you mentions actually backs this up to 100%, but this is already something done for many many years now.